### PR TITLE
[13.0][ADD] base_recurrence

### PR DESCRIPTION
**Pull request by rousseldenis**
This idea came from : https://github.com/OCA/contract/blob/14.0/contract/models/contract_recurrency_mixin.py#L141

where I see a perspective of duplicate code in other modules.

### CODE REVIEW

**Comment by rousseldenis** [comment]
Example of use : https://github.com/OCA/project/pull/839

**Comment by rousseldenis** [comment]
@OCA/tools-maintainers

File: base_recurrence/readme/DESCRIPTION.rst
Comment (pedrobaeza) [inline base_recurrence/readme/DESCRIPTION.rst]:
```suggestion
It allows to define a recurring rule on a record and two helper methods:

* One to get the recurring next date
* One to set the recurring next date on the record
```

File: base_recurrence/readme/USAGE.rst
Comment (pedrobaeza) [inline base_recurrence/readme/USAGE.rst]:
```suggestion
Supported intervals
~~~~~~~~~~~~~~~~~~~
```

File: base_recurrence/readme/USAGE.rst
Comment (pedrobaeza) [inline base_recurrence/readme/USAGE.rst]:
```suggestion
Example
~~~~~~~
```

**Comment by rousseldenis** [comment]
@pedrobaeza Thanks for your comments. I will attend them

**Comment by thomaspaulb** [comment]
Is there some overlap with this?

https://github.com/OCA/server-tools/pull/2356/files

**Comment by rousseldenis** [comment]
> Is there some overlap with this?
> 
> https://github.com/OCA/server-tools/pull/2356/files

Mmhh, maybe but with very different implentations.

We can maybe discuss of a convergence on OCA days if you come ?

**Review by pedrobaeza — APPROVED** [review]
Please squash before merging.

**Pull request closed without merging**
